py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,512 @@
+from oauth2client import client
+from oauth2client import clientsecrets
+from oauth2client import tools
+from oauth2client.file import Storage
+from httplib2 import ServerNotFoundError
+
+
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+APPLICATION_NAME = 'py3status google_calendar module'
+
+
+class Py3status:
+    auth_token = '~/.config/py3status/google_calendar.auth_token'
+    button_open = 3
+    button_refresh = 2
+    button_toggle = 1
+    cache_timeout = 60
+    client_secret = '~/.config/py3status/google_calendar.client_secret'
+    force_lowercase = False
+    format = '{events}| \?color=event \u2687'
+    format_date = '%a %d-%m'
+    format_event = '\?color=event {summary}[\?if=location  ({location})][\?if=time_to  {time_to}]'
+    format_event_full = '[\?color=event {summary}] ' +\
+        '[\?color=time ({start_time} - {end_time}, {start_date})]'
+    format_separator = '\?color=separator \|'
+    format_time = '%I:%M %p'
+    format_time_left = '\?color=time ([\?if=days {days}d ]' +\
+        '[\?if=hours {hours}h ][\?if=minutes {minutes}m]) left'
+    format_time_to = '\?color=time ([\?if=days {days}d ]' +\
+        '[\?if=hours {hours}h ][\?if=minutes {minutes}m])'
+    ignore_all_day_events = False
+    num_events = 3
+    thresholds = []
+    time_to_max = 180
+    warn_threshold = 0
+    warn_timeout = 300
+
+    def post_config_hook(self):
+        self.button = None
+        self.button_states = [False] * self.num_events
+        self.events = None
+        self.no_update = False
+
+        self.client_secret = os.path.expanduser(self.client_secret)
+        self.auth_token = os.path.expanduser(self.auth_token)
+
+        self.credentials = self._get_credentials()
+        self.is_authorized = self._authorize_credentials()
+
+    def _get_credentials(self):
+        """
+        Gets valid user credentials from storage.
+
+        If nothing has been stored, or if the stored credentials are invalid,
+        the OAuth2 flow is completed to obtain the new credentials.
+
+        Returns: Credentials, the obtained credential.
+        """
+        client_secret_path = os.path.dirname(self.client_secret)
+        auth_token_path = os.path.dirname(self.auth_token)
+
+        if not os.path.exists(auth_token_path):
+            os.makedirs(auth_token_path)
+
+        if not os.path.exists(client_secret_path):
+            os.makedirs(client_secret_path)
+
+        flags = tools.argparser.parse_args(args=[])
+        store = Storage(self.auth_token)
+        credentials = store.get()
+
+        if not credentials or credentials.invalid:
+            try:
+                flow = client.flow_from_clientsecrets(self.client_secret, SCOPES)
+                flow.user_agent = APPLICATION_NAME
+                if flags:
+                    credentials = tools.run_flow(flow, store, flags)
+                else:  # Needed only for compatibility with Python 2.6
+                    credentials = tools.run(flow, store)
```

Comment:
(Opinion) Test this module on `python2` too... and without the `else` too as maybe we don't need this anymore. 
```
which py3status
cd  #THE_PATH
sudo cp py3status py3status2
sudo $EDITOR py3status2
change python to python2 in first line

change i3 config... py3status -c... to py3status2...
```
or sometimes in the terminal...
```
python2 google_calendar.py
```

Reply:
Hmm, trying the first approach I get:
```
Traceback (most recent call last):
  File "/usr/bin/py3status", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3142, in <module>
    @_call_aside
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3126, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3155, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 664, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 981, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 867, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'py3status' distribution was not found and is required by the application
```

Which seems similar to #122, but I installed it using `setup.py` so not sure here...

Trying `python2 google_calendar.py`, I get 

```
Traceback (most recent call last):
  File "google_calendar.py", line 520, in <module>
    from py3status.module_test import module_test
ImportError: No module named py3status.module_test
```

Which I don't think is related to my module.

Reply:
Try `sudo python3 develop install` and `sudo python2 develop install`.